sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1607,7 +1607,9 @@ class Dataset[T] private[sql](
    */
   @Experimental
   @InterfaceStability.Evolving
-  def reduce(func: (T, T) => T): T = rdd.reduce(func)
+  def reduce(func: (T, T) => T): T = withNewExecutionId {
```

Comment:
this method should use `withNewRDDExecutionId`

Reply:
Thanks, that makes sense when I looked at the code for `foreach` and `foreachPartition`; I put up a new version with this change. It however wasn't clear immediately how the new function `withNewRDDExecutionId` would be beneficial over `withNewExecutionId`, can you elaborate a little when you get the chance?

Reply:
@sohama4 Spark Web UI will show no SQL metrics when using `withNewExecutionId`. See https://github.com/apache/spark/pull/19689